File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    int n = 0;
    
    for(int i=0;i<s.size();i++){
        if(n < 0) return false;
        
        if(s[i]=='(')
            n++;
        else
            n--;
    }

    return n==0;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let count = 0;
    let answer = true;
    
    for(let i=0; i<s.length; i++) {
         if (s[i] === ')' && !(count > 0))  {
            answer = false;
         }
         
         else {
             (s[i] === '(') && count++;
             (s[i] === ')') && count--;
         }     
     }
    
    answer =  count === 0 && answer===true ?  true : false;
    return answer;
}